Add login logic

handymenny opened the pull request.

handymenny committed:
add login logic

veetaw commented on src/main/java/com/glucoseguardian/webbackend/application/adapter/LoginAdapter.java:
```diff
@@ -0,0 +1,78 @@
+import com.glucoseguardian.webbackend.storage.entity.Paziente;
+import com.glucoseguardian.webbackend.storage.entity.Tutore;
+
+public class LoginAdapter implements LoginAdapterInterface {
```
Since the class handles login, logout, password change and I think in future registration (?), shouldn't be AuthAdapter a better name?

veetaw commented on src/main/java/com/glucoseguardian/webbackend/application/adapter/LoginAdapter.java:
```diff
@@ -0,0 +1,78 @@
+    String idUtente;
+    int tipoUtente;
+
+    Admin adminRes = adaptee.loginAdmin(email, password, otp);
```
Maybe we should have a UserDAO with a findColumnByEmail that returns if the searched email is connected to a Tutore, an Admin, a Paziente or a Dottore; or we should separate the different login, this is bad. (at least separate the admin and dottore ones who will be used in completely separate scopes).

veetaw commented on src/main/java/com/glucoseguardian/webbackend/application/service/LoginService.java:
```diff
@@ -0,0 +1,89 @@
+  private TutoreDao tutoreDao;
+
+  @Override
+  public Admin loginAdmin(String email, String password, Integer otp) {
```
Maybe return `Optional<Admin>` to give the code a modern look?

veetaw commented on src/main/java/com/glucoseguardian/webbackend/application/service/LoginService.java:
```diff
@@ -0,0 +1,89 @@
+        return entity;
+      }
+    }
+    // TODO: Add a specific exception instead of Null
```
If we return `Optional<Paziente>` an exception will be unnecessary

veetaw commented on src/main/java/com/glucoseguardian/webbackend/storage/dao/AdminDao.java:
```diff
@@ -1,11 +1,14 @@
  */
 public interface AdminDao extends JpaRepository<Admin, String> {
 
+  Optional<Admin> findByEmail(String email);
```
lgtm

veetaw commented on src/main/java/com/glucoseguardian/webbackend/storage/dto/AssunzioneFarmacoDto.java:
```diff
@@ -10,17 +10,17 @@
-  private long id;
-  private long idFarmaco;
-  private int dosaggio;
+  private Long id;
```
Isn't it better to also use wrapper classes around primitives in the daos? In order to prevent a lot of autoboxing and unboxings?

veetaw replied:
(same for all the other dto)

veetaw commented on src/main/java/com/glucoseguardian/webbackend/application/adapter/LoginAdapter.java:
```diff
@@ -0,0 +1,78 @@
+    Admin adminRes = adaptee.loginAdmin(email, password, otp);
+    if (adminRes != null) {
+      idUtente = adminRes.getCodiceFiscale();
+      tipoUtente = 0;
```
I think it's time to have an enum for this, with a toDBInteger function and a fromDBInteger function, it's unclear expressed like this

veetaw commented on src/main/java/com/glucoseguardian/webbackend/application/adapter/LoginAdapter.java:
```diff
@@ -0,0 +1,78 @@
+      return new LoginOutputDto(idUtente, tipoUtente, null);
+    }
+
+    Dottore dottoreRes = adaptee.loginDottore(email, password, otp);
```
If we want to use this kind of "architecture" shouldn't be at least a bit better to make Dottore, Paziente, etc to implement an interface that has a login method? since it's the same for everything. Maybe we can use it like:

```java
Optional<Dottore> dottoreOpt = (Optional<Dottore>) adaptee.login(email, password, otp);
LoginOutputDto res = null;
dottoreOpt.ifPresent(dottore -> res = LoginOutputDto(dottore.getCodiceFiscale, Dottore.TIPO_UTENTE, null));
```

veetaw replied:
I don't know if that makes sense tbh lol

veetaw reviewed (changes requested):
see comments, just suggestions for the architecture, np for me if we want to keep it like this, sometime KISS is the best

handymenny commented:
Rebased on top of main and implemented bridge pattern.
The only revelant commit is 1bc87a84ed33393f7d8482a95792b978450274f2 
The others will be sent in different pull requests

@veetaw @Scient122 @AmineSr97 waiting for your feedback

handymenny committed:
add stub and annotations

handymenny committed:
update auth logic
